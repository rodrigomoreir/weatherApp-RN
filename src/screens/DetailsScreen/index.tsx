import React from 'react';
import { useNavigation, useRoute } from '@react-navigation/native';

import { MotiView } from 'moti';

import { Header } from '../../components/Header';
import { CityProps } from '../HomeScreen';

import {
  StyledContainer,
  StyledWeatherContent,
  StyledCityTitle,
  StyledTemperatureTitle,
  StyledWeatherTitle,
  StyledWeatherDetailsCard,
  StyledSeparator,
  StyledInfoContent
} from './styles';

interface Params {
  cityData: CityProps
}

const DetailsScreen = () => {
  const { goBack } = useNavigation();
  const route = useRoute();
  const { cityData } = route.params as Params

  return (
    <>
      <Header title={'Details'} goBack={goBack} />
      <StyledContainer>
        <StyledWeatherContent>
          <StyledCityTitle>{cityData.name} - {cityData.country}</StyledCityTitle>
          <StyledTemperatureTitle>{cityData.temperature}°</StyledTemperatureTitle>
          <StyledWeatherTitle>{cityData.weather}</StyledWeatherTitle>
          <StyledWeatherTitle>Max.: {cityData.temperatureMax}° | Min.: {cityData.temperatureMin}°</StyledWeatherTitle>
          <StyledWeatherTitle>Feels like: {cityData.feelsLike}°</StyledWeatherTitle>
        </StyledWeatherContent>
        <MotiView>

          <StyledWeatherDetailsCard>

            <StyledWeatherTitle>Outras informações de Clima</StyledWeatherTitle>
            <StyledSeparator />

            <StyledInfoContent>
              <StyledWeatherTitle>Visibility</StyledWeatherTitle>
              <StyledWeatherTitle>{cityData.visibility}</StyledWeatherTitle>
            </StyledInfoContent>

            <StyledInfoContent>
              <StyledWeatherTitle>Humidity</StyledWeatherTitle>
              <StyledWeatherTitle>{cityData.humidity} %</StyledWeatherTitle>
            </StyledInfoContent>

            <StyledInfoContent>
              <StyledWeatherTitle>Speed Wind</StyledWeatherTitle>
              <StyledWeatherTitle>{cityData.speedWind} m/s</StyledWeatherTitle>
            </StyledInfoContent>

            <StyledInfoContent>
              <StyledWeatherTitle>Cloud Percentage</StyledWeatherTitle>
              <StyledWeatherTitle>{cityData.cloudPercentage} %</StyledWeatherTitle>
            </StyledInfoContent>

            <StyledInfoContent>
              <StyledWeatherTitle>Atmospheric Pressure</StyledWeatherTitle>
              <StyledWeatherTitle>{cityData.atmPressure} hPa</StyledWeatherTitle>
            </StyledInfoContent>

          </StyledWeatherDetailsCard>
        </MotiView>
      </StyledContainer>
    </>
  );
}

export default DetailsScreen;
