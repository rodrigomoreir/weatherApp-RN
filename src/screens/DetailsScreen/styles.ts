import styled from 'styled-components/native';

export const StyledContainer = styled.View`
  flex: 1;
  padding: 0 15px;
  background-color: ${({ theme }) => theme.colors.background};
`

export const StyledWeatherContent = styled.View`
  padding: 15px;
  align-items: center;
  justify-content: center;
`

export const StyledCityTitle = styled.Text`
  font-size: 30px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.textPrimary};
`

export const StyledTemperatureTitle = styled.Text`
  font-size: 65px;
  color: ${({ theme }) => theme.colors.textPrimary};
`

export const StyledWeatherTitle = styled.Text`
  font-size: 18px;
  margin-bottom: 3px;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.textPrimary};
`

export const StyledWeatherDetailsCard = styled.View`
  margin-top: 10px;
  padding: 15px;
  border-radius: 20px;
  border-width: 0.5px;
  border-color: ${({ theme }) => theme.colors.border};
  background-color: ${({ theme }) => theme.colors.card};
`

export const StyledSeparator = styled.View`
  height: 1px;
  border: 1px;
  border-color: ${({ theme }) => theme.colors.border};
  margin: 10px 0;
`

export const StyledInfoContent = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin-top: 5px;
`
