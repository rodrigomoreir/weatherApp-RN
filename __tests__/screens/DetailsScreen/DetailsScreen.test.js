import React from 'react'
import DetailsScreen from '../../../src/screens/DetailsScreen'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/helpers/testing'

describe('Details Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<DetailsScreen />)
  })

  test('renders correctly', () => {
    renderWithTheme(<DetailsScreen />)
  })
})
