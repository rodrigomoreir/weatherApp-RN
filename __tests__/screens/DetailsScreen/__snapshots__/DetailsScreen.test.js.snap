// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Screen matches snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": 10,
          },
        ]
      }
    >
      Details
    </Text>
  </View>,
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 0,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View>
      <View
        animate={
          Object {
            "opacity": 1,
            "translateY": 0,
          }
        }
        from={
          Object {
            "opacity": 0,
            "translateY": -50,
          }
        }
        style={
          Array [
            Object {
              "opacity": [MockFunction],
              "transform": [MockFunction],
            },
          ]
        }
        transition={
          Object {
            "duration": 700,
            "type": "timing",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                  "fontWeight": "bold",
                },
              ]
            }
          >
             - 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 65,
                },
              ]
            }
          >
            °C
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 3,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 3,
                },
              ]
            }
          >
            Max.: 
            ° | Min.: 
            °
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 3,
                },
              ]
            }
          >
            Feels like: 
            °
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F5F8FA",
                "borderColor": "#808080",
                "borderRadius": 20,
                "borderWidth": 0.5,
                "marginBottom": 10,
                "marginTop": 10,
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 3,
                },
              ]
            }
          >
            Outras informações de Clima
          </Text>
          <View
            style={
              Array [
                Object {
                  "borderColor": "#808080",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "height": 1,
                  "marginBottom": 10,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 10,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
              Visibility
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
              Humidity
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
               %
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
              Speed Wind
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
               m/s
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
              Cloud Percentage
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
               %
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 5,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
              Atmospheric Pressure
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 3,
                  },
                ]
              }
            >
               hPa
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        <View
          autoPlay={true}
          cacheComposition={true}
          collapsable={false}
          enableMergePathsAndroidForKitKatAndAbove={false}
          forwardedRef={[Function]}
          loop={true}
          nativeID="animatedComponent"
          onAnimationFinish={[Function]}
          onLayout={[Function]}
          progress={0}
          resizeMode="contain"
          speed={1}
          style={
            Object {
              "alignSelf": "center",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>,
]
`;
